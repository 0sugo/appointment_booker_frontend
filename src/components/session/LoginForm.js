import React, { useState } from 'react';
import { Link, useNavigate } from 'react-router-dom';

const LoginForm = () => {
  const [formData, setFormData] = useState({
    email: '',
    password: '',
  });

  const [loginSuccess, setLoginSuccess] = useState(false);
  const [loginFailure, setLoginFailure] = useState(false);

  const navigate = useNavigate();

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((prevData) => ({
      ...prevData,
      [name]: value,
    }));
  };

  const url = 'http://127.0.0.1:4000/api/v1/users/sign_in';
  const handleSubmit = async (e) => {
    e.preventDefault();

    try {
      const response = await fetch(url, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({ user: formData }),
      });

      if (response.ok) {
        const data = await response.json();
        localStorage.setItem('userData', JSON.stringify(data));
        setLoginSuccess(true);
        navigate('/homepage');
      } else {
        setLoginFailure(true);
      }
    } catch (error) {
      throw new Error(error);
    }
  };

  return (
    <div className="splash_page">
      <h2 className="h2_login">Login</h2>
      {loginSuccess ? (
        <div>
          <p>Login successful! Welcome back.</p>
        </div>
      ) : (
        <form onSubmit={handleSubmit}>
          <label htmlFor="email">
            Email:
            <br />
            <input
              id="email"
              type="email"
              name="email"
              value={formData.email}
              onChange={handleChange}
            />
          </label>
          <br />
          <label htmlFor="password">
            Password:
            <br />
            <input
              id="password"
              type="password"
              name="password"
              value={formData.password}
              onChange={handleChange}
            />
          </label>
          <br />
          <button type="submit" className="splash_link login_btn">Login</button>
          <Link to="/register" className="splash_link">Register</Link>
        </form>

      )}

      {loginFailure && (
      <div>
        <p className="login_error">Invalid email or password.</p>
      </div>
      )}
    </div>

  );
};

export default LoginForm;
