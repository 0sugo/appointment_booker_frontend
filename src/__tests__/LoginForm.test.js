import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import LoginForm from '../components/session/LoginForm';

describe('LoginForm', () => {
  test('renders correctly', () => {
    const loginPage = render(
      <Router>
        <LoginForm />
      </Router>,
    );
    expect(loginPage).toMatchSnapshot();
  });
});
