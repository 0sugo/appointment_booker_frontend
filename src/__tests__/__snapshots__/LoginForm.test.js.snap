// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="splash_page"
      >
        <h2
          class="h2_login"
        >
          Login
        </h2>
        <form>
          <label
            for="email"
          >
            Email:
            <br />
            <input
              id="email"
              name="email"
              type="email"
              value=""
            />
          </label>
          <br />
          <label
            for="password"
          >
            Password:
            <br />
            <input
              id="password"
              name="password"
              type="password"
              value=""
            />
          </label>
          <br />
          <button
            class="splash_link login_btn"
            type="submit"
          >
            Login
          </button>
          <a
            class="splash_link"
            href="/register"
          >
            Register
          </a>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="splash_page"
    >
      <h2
        class="h2_login"
      >
        Login
      </h2>
      <form>
        <label
          for="email"
        >
          Email:
          <br />
          <input
            id="email"
            name="email"
            type="email"
            value=""
          />
        </label>
        <br />
        <label
          for="password"
        >
          Password:
          <br />
          <input
            id="password"
            name="password"
            type="password"
            value=""
          />
        </label>
        <br />
        <button
          class="splash_link login_btn"
          type="submit"
        >
          Login
        </button>
        <a
          class="splash_link"
          href="/register"
        >
          Register
        </a>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
